### ISSUE TITLE
SROS - Commit triggers NetmikoTimeoutException

### ISSUE THREAD

**Issue by jtishey** [ktbyers/netmiko#1663]
I'm seeing an issue where I use the Nokia SROS driver to connect to a VM on SROS 16.0R7 in MD-CLI mode.   When I call the commit() method after send_config_set() - the configuration on the router commits, but netmiko raises a timeout: `netmiko.ssh_exception.NetmikoTimeoutException: Timed-out reading channel, data not available.`

Python 3.6.7
netmiko==3.1.0
TiMOS-C-16.0.R7-1 cpm/x86_64 Nokia 7750 SR
configuration-mode model-driven

@wisotzky - have you seen anything similar working on #1442  ?

Example:
```
>>> from netmiko import ConnectHandler
>>> d = {
...  'device_type': 'nokia_sros',
...  'ip': 'router1',
...  'username': username,
...  'password': password,
...  'port': 22,
...  'verbose': False
... }
>>> nc = ConnectHandler(**d)
>>> nc.send_config_set(["configure port 1/1/c10 description test"])
'edit-config exclusive\nINFO: CLI #2060: Entering exclusive configuration mode\nINFO: CLI #2061: Uncommitted changes are discarded on configuration mode exit\n\n(ex)[]\nA:user@router1# configure port 1/1/c10 description test\n\n*(ex)[]\nA:user@router1# '
>>> nc.commit()
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/paramiko/channel.py", line 699, in recv
    out = self.in_buffer.read(nbytes, self.timeout)
  File "/usr/lib/python3.6/site-packages/paramiko/buffered_pipe.py", line 164, in read
    raise PipeTimeout()
paramiko.buffered_pipe.PipeTimeout

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/netmiko/base_connection.py", line 550, in _read_channel_expect
    new_data = self.remote_conn.recv(MAX_BUFFER)
  File "/usr/lib/python3.6/site-packages/paramiko/channel.py", line 701, in recv
    raise socket.timeout()
socket.timeout

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.6/site-packages/netmiko/nokia/nokia_sros_ssh.py", line 128, in commit
    output += self.read_until_pattern(r"@")
  File "/usr/lib/python3.6/site-packages/netmiko/base_connection.py", line 627, in read_until_pattern
    return self._read_channel_expect(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/netmiko/base_connection.py", line 561, in _read_channel_expect
    "Timed-out reading channel, data not available."
netmiko.ssh_exception.NetmikoTimeoutException: Timed-out reading channel, data not available.
>>>
```

The config does get committed, so either it's just not picking up the return afterwards, or it's taking too long - which I think is unlikely because it commits almost instantly via CLI.

Example of the same change via CLI
```

[]
A:user@router1# edit-config exclusive
INFO: CLI #2060: Entering exclusive configuration mode
INFO: CLI #2061: Uncommitted changes are discarded on configuration mode exit

(ex)[]
A:user@router1# configure port 1/1/c10 description test123

*(ex)[]
A:user@router1# commit

(ex)[]
A:user@router1# quit-config
INFO: CLI #2064: Exiting exclusive configuration mode
```

From nokia/nokia_sros_ssh.py
```
    def commit(self, *args, **kwargs):
        """Activate changes from private candidate for Nokia SR OS"""
        output = self._exit_all()
        if "@" in self.base_prompt and "*(ex)[" in output:
            log.info("Apply uncommitted changes!")
            cmd = "commit"
            self.write_channel(self.normalize_cmd(cmd))
            output += self.read_until_pattern(pattern=re.escape(cmd))
            output += self.read_until_pattern(r"@")
        return output
```

Any help is appreciated.

**Comment by jtishey** [comment]
I think having two read_until_pattern calls in a row is causing the issue.  Maybe on longer commits it's ok, but I think the VM with small config changes is returning output too fast possibly?  If I comment out the second read_until_pattern it seems to work normally.

nokia_sros_ssh.py
```            cmd = "commit"
            self.write_channel(self.normalize_cmd(cmd))
            output += self.read_until_pattern(pattern=re.escape(cmd))
            #output += self.read_until_pattern(r"@")
```


Sample with `output += self.read_until_pattern(r"@")`:
```
>>> nc.send_config_set(['configure port 1/1/c10 description "test5"'])
'edit-config exclusive\nINFO: CLI #2060: Entering exclusive configuration mode\nINFO: CLI #2061: Uncommitted changes are discarded on configuration mode exit\n\n(ex)[]\nA:user@router1# configure port 1/1/c10 description "test5"\n\n*(ex)[]\nA:user@router1# '
>>>
>>> nc.commit()
'exit all\r\n\r\n*(ex)[]\r\nA:user@router1# commit\r\n\r\n(ex)[]\r\nA:user@router1# '
>>>
>>> nc.check_config_mode()
True
>>>
>>> nc.exit_config_mode()
'exit all\r\n\r\n(ex)[]\r\nA:user@router1# quit-config\r\nINFO: CLI #2064: Exiting exclusive configuration mode\r\n\r\n[]\r\nA:user@router1# '
>>>
>>> nc.check_config_mode()
False
```

**Comment by jtishey** [comment]
Wil try changing the pattern to "^(ex)[]$"

**Comment by jtishey** [comment]
Updated pattern to r"[^\*]\(ex\)\[]"

Tested on physical SR7s 16.0.R12 as well and this seems to work.

**Comment by ktbyers** [comment]
Should be fixed now.

**Comment by hellt** [comment]
@ktbyers has this been fixed by #1672 ?

**Comment by jtishey** [comment]
@hellt  that is correct
